modules/core/src/core/animation-loop.js

```diff
@@ -73,6 +75,8 @@ export default class AnimationLoop {
     this._onMousemove = this._onMousemove.bind(this);
     this._onMouseleave = this._onMouseleave.bind(this);
 
+    this._onFirstStart = runOnce(this._onFirstStart.bind(this));
```

Comment:
I think this makes it unclear what `onFirstStart` is doing. If you look at the method and where it's used, it looks like it's being called every time `start` is called. Plus, it doesn't look like luma handles context loss right now, but when we do implement it, this will make it harder to set things up again.

I'd suggest just having an `initialized` flag that `_onFirstStart` checks and just returns if it's set.

Reply:
+1

Reply:
Makes sense to me! How about I rename `_onFirstStart` to `_initialize` and add an `_initialized` flag, as you suggest?

Reply:
Actually, I'm going to put that `_onFirstStart` logic back into `start` to reduce the number of changes here.